src/textual/pilot.py

```diff
@@ -139,10 +179,21 @@ async def hover(
         message_arguments = _get_mouse_message_arguments(
             target_widget, offset, button=0
         )
+
+        hover_offset = Offset(message_arguments["x"], message_arguments["y"])
+        visible_screen_region = screen.region + screen.scroll_offset
```

Comment:
I don't think this is correct. Which suggests the tests aren't doing their job.

Reply:
Of course this isn't correct. It should match `Pilot.click`. See 7d11422.